acceptance/tests/augeas_validation.rb

```diff
@@ -0,0 +1,42 @@
+skip_test 'non-windows only test' if hosts.any? { |host| host.platform =~ /windows/ }
+tag 'audit:high'
+test_name 'Augeas Validation' do
+  teardown do
+    hosts.each do |agent|
+      file = <<-EOF
+augeas { 'test_ssh':
+  lens    => 'Ssh.lns',
+  incl    => '/etc/ssh/ssh_config',
+  context => '/files/etc/ssh/ssh_config',
+  changes => [
+    'remove Host testing.testville.nil'
+  ]
+}
+EOF
+      create_remote_file(agent, '/tmp/augeas_validation_teardown.pp', file)
+      on(agent, '/opt/puppetlabs/puppet/bin/puppet apply /tmp/augeas_validation_teardown.pp')
+      on(agent, 'rm /tmp/augeas_validation.pp')
+      on(agent, 'rm /tmp/augeas_validation_teardown.pp')
+    end
+  end
+
+  hosts.each do |agent|
+    step 'validate Augeas binary' do
+      on(agent, '/opt/puppetlabs/puppet/bin/augtool --version')
+    end
+    step 'validate we can apply a resource type augeas' do
+      file = <<-EOF
+augeas { 'test_ssh':
+  lens    => 'Ssh.lns',
+  incl    => '/etc/ssh/ssh_config',
+  context => '/files/etc/ssh/ssh_config',
+  changes => [
+    'set Host testing.testville.nil'
+  ]
+}
+EOF
+      create_remote_file(agent, '/tmp/augeas_validation.pp', file)
+      on(agent, '/opt/puppetlabs/puppet/bin/puppet apply /tmp/augeas_validation.pp')
```

Comment:
Should we have an assert here? It feels a little bit unclear that we are just relying on the apply not returning a non-zero exit code as the test, but I could really go either way.

Also, instead creating a remote file, you could try inlining the code so that it reduces some network traffic? I tested this on a pooler instance and it looks to work fine:
```
[4] pry(#<Beaker::TestCase>)> on(agent, "puppet apply -e \"#{file}\"")

maroon-elevator.delivery.puppetlabs.net (maroon-elevator.delivery.puppetlabs.net) 12:11:11$ puppet apply -e "augeas { 'test_ssh':
  lens    => 'Ssh.lns',
  incl    => '/etc/ssh/ssh_config',
  context => '/files/etc/ssh/ssh_config',
  changes => [
    'set Host testing.testville.nil'
  ]
}
"
  Notice: Compiled catalog for maroon-elevator.delivery.puppetlabs.net in environment production in 0.06 seconds
  Notice: /Stage[main]/Main/Augeas[test_ssh]/returns: executed successfully
  Notice: Applied catalog in 0.23 seconds
```